src/model/CoilCoolingDX_Impl.hpp

```diff
@@ -0,0 +1,156 @@
+
+  /** CoilCoolingDX_Impl is a StraightComponent_Impl that is the implementation class for CoilCoolingDX.*/
+  class MODEL_API CoilCoolingDX_Impl : public StraightComponent_Impl {
+   public:
+    /** @name Constructors and Destructors */
+    //@{
+
+    CoilCoolingDX_Impl(const IdfObject& idfObject,
+                       Model_Impl* model,
+                       bool keepHandle);
+
+    CoilCoolingDX_Impl(const openstudio::detail::WorkspaceObject_Impl& other,
+                       Model_Impl* model,
+                       bool keepHandle);
+
+    CoilCoolingDX_Impl(const CoilCoolingDX_Impl& other,
+                       Model_Impl* model,
+                       bool keepHandle);
+
+    virtual ~CoilCoolingDX_Impl() {}
+
+    //@}
+    /** @name Virtual Methods */
+    //@{
+
+    virtual const std::vector<std::string>& outputVariableNames() const override;
+
+    virtual IddObjectType iddObjectType() const override;
+
+    virtual std::vector<ScheduleTypeKey> getScheduleTypeKeys(const Schedule& schedule) const override;
+
+    //@}
+    /** @name Getters */
+    //@{
+
+    boost::optional<Schedule> availabilitySchedule() const;
+
+    boost::optional<ThermalZone> condenserZone() const;
+
+    boost::optional<std::string> condenserInletNodeName() const;
+
+    boost::optional<std::string> condenserOutletNodeName() const;
+
+    CoilCoolingDXCurveFitPerformance performanceObject() const;
+
+    boost::optional<std::string> condensateCollectionWaterStorageTankName() const;
+
+    boost::optional<std::string> evaporativeCondenserSupplyWaterStorageTankName() const;
+
+    //@}
+    /** @name Setters */
+    //@{
+
+    bool setAvailabilitySchedule(Schedule& schedule);
+
+    void resetAvailabilitySchedule();
+
+    bool setCondenserZone(const ThermalZone& thermalZone);
+
+    void resetCondenserZone();
+
+    bool setCondenserInletNodeName(const std::string& condenserInletNodeName);
+
+    void resetCondenserInletNodeName();
+
+    bool setCondenserOutletNodeName(const std::string& condenserOutletNodeName);
+
+    void resetCondenserOutletNodeName();
+
+    bool setPerformanceObject(const CoilCoolingDXCurveFitPerformance& coilCoolingDXCurveFitPerformance);
+
+    bool setCondensateCollectionWaterStorageTankName(const std::string& condensateCollectionWaterStorageTankName);
+
+    void resetCondensateCollectionWaterStorageTankName();
+
+    bool setEvaporativeCondenserSupplyWaterStorageTankName(const std::string& evaporativeCondenserSupplyWaterStorageTankName);
+
+    void resetEvaporativeCondenserSupplyWaterStorageTankName();
+
+    //@}
+    /** @name Other */
+    //@{
+
+    virtual unsigned inletPort() const override;
+
+    virtual unsigned outletPort() const override;
+
+    virtual ModelObject clone(Model model) const override;
+
+    virtual std::vector<ModelObject> children() const override;
+
+    virtual boost::optional<HVACComponent> containingHVACComponent() const override;
+
+    virtual bool addToNode(Node & node) override;
```

Comment:
Move above in virtual section.

Reply:
Perhaps add `    virtual std::vector<IddObjectType> allowableChildTypes() const override;` ... ?